### PR TITLE
Shift eval iterations to run after the first set of training iterations.

### PR DESCRIPTION
**Pull request by arthurarg**

### CODE REVIEW

File: tests/dopamine/discrete_domains/run_async_training_test.py [hunk 1]
```diff
@@ -111,7 +111,15 @@ def _put_call_cnt(v):
     self.assertEqual(_put_call_cnt(None), 2)  # Stop task.
 
   def testNumberSteps(self):
```
Comment (wes-turner) [inline tests/dopamine/discrete_domains/run_async_training_test.py:115]: Suggsetion here to separate each of these checks into its own test.

File: dopamine/discrete_domains/run_experiment.py [hunk 2]
```diff
@@ -608,10 +608,12 @@ def _run_iterations(self):
     # attribute.
     self._completed_iteration = self._start_iteration
     for iteration in range(self._start_iteration, self._num_iterations):
-      if (iteration + 1) % self._eval_period == 0:
+      if iteration and iteration % self._eval_period == 0:
```
Comment (wes-turner) [inline dopamine/discrete_domains/run_experiment.py:613]: This isn't exactly the right time or place to comment, but do you think `self._eval_period` is a good choice for an argument to the `AsyncRunner` constructor?

**Review by wes-turner — APPROVED** [review]
LGTM

File: dopamine/discrete_domains/run_experiment.py (hunk 2 above)
Comment (arthurarg) [reply dopamine/discrete_domains/run_experiment.py:613]: I think it is a good choice. We can definitely add it.

Comment (arthurarg) [reply dopamine/discrete_domains/run_experiment.py:613]: I added a TODO for that.

File: tests/dopamine/discrete_domains/run_async_training_test.py (hunk 1 above)
Comment (arthurarg) [reply tests/dopamine/discrete_domains/run_async_training_test.py:115]: done.

**Pull request merged**